spec/datadog/tracing/integration_spec.rb

```diff
@@ -619,24 +716,19 @@ def sample!(trace)
     end
 
     it do
-      3.times do |i|
-        tracer.trace('parent_span') do
-          tracer.trace('child_span') do |_span, trace|
-            # I want to keep the trace to which `child_span` belongs
-            trace.sampling_priority = i
-          end
-        end
+      tracer.trace('parent_span') do
+        tracer.trace('child_span') {}
+      end
 
-        try_wait_until(attempts: 20) { tracer.writer.stats[:traces_flushed] >= 1 }
-        stats = tracer.writer.stats
+      try_wait_until { tracer.writer.stats[:traces_flushed] >= 1 }
+      stats = tracer.writer.stats
 
-        expect(stats[:traces_flushed]).to eq(1)
-        expect(stats[:transport].client_error).to eq(0)
-        expect(stats[:transport].server_error).to eq(0)
-        expect(stats[:transport].internal_error).to eq(0)
+      expect(stats[:traces_flushed]).to eq(1)
+      expect(stats[:transport].client_error).to eq(0)
+      expect(stats[:transport].server_error).to eq(0)
+      expect(stats[:transport].internal_error).to eq(0)
 
-        expect(out).to have_received(:puts)
-      end
+      expect(out).to have_received(:puts)
```

Comment:
This test created 3 traces but didn't wait enough time for the Writer thread to flush them all.

Then, it waited for exactly 1 of the traces to be flushed and asserted 3 times, in quick succession, that only 1 span was flushed.
Modifying this test by adding a `sleep` in the iteration shows that all 3 spans will be flushed normally.

I simplified the test, as the 3-time assertions don't add value.
The only difference between the spans is the `trace.sampling_priority = i`, but sampling priority won't affect this code path (all spans will be flushed at this point, regardless of sampling priority).

This is also a backport from the Sampling Decision Mechanism feature.

It's backported here because it broke CI in this branch 😢

Reply:
👍  I very much like this change :)